libflux/src/semantic/tests.rs

```diff
@@ -204,6 +204,636 @@ fn instantiation_1() {
     }
 }
 #[test]
+fn instantation_literals() {
+    test_infer! {
+        src: r#"
+            a = "Hello, World!"
+            b = 12
+            c = 18.5
+            d = 12h
+            e = 2019-10-31T00:00:00Z
+            f = /server[01]/
+        "#,
+        exp: &[
+            ("a", "forall [] string"),
+            ("b", "forall [] int"),
+            ("c", "forall [] float"),
+            ("d", "forall [] duration"),
+            ("e", "forall [] time"),
+            ("f", "forall [] regexp"),
+        ],
+    }
+}
+#[test]
+fn instantiate_string_expr() {
+    let src = r#"message = "Hello, ${name}!""#;
+    test_infer! {
+        env: &[
+            ("name", "forall [t0] t0"),
+        ],
+        src: src,
+        exp: &[
+            ("message", "forall [] string"),
+        ],
+    }
+    test_infer_err! {
+        env: &[
+            ("name", "forall [] int"),
+        ],
+        src: src,
+    }
```

Comment:
I like this test case. Can you add more for the other basic non-string datatypes?